Offline coupling with Delwaq (#1137)
Deltares/Ribasim: coupling/delwaq/README.md

# Ribasim Delwaq coupling

## Steps

> Choose a testmodel and run it. For example, running the basic model with ribasim_cli:

Reviewer: ```suggestion
Choose a model and run it. For example, running the basic model with the Ribasim CLI:
```

```bash

> ./build/ribasim_cli/bin/ribasim generated_testmodels/basic/ribasim.toml

Reviewer: ```suggestion
ribasim generated_testmodels/basic/ribasim.toml
```

The `./build/ribasim_cli/bin/ribasim` path is outdated.

```

> Run `gen_delwaq.py` to couple the models. Open a terminal in the directory of Ribasim

Reviewer: ```suggestion
Run `generate.py` to generate a Delwaq model. Open a terminal in the Ribasim repository root and run:
```

> ```python
> python ./coupling/delwaq/gen_delwaq.py

Reviewer: ```suggestion
```bash
python ./coupling/delwaq/generate.py
```

```
The script can:
- Open an existing Ribasim model (hardcoded to `generated_testmodels/basic/ribasim.toml` for now)
- Convert the metadata, topology and substances to Delwaq format and put it into the `model` folder

> You now should manually run Delwaq. You can do so with the Docker image.

Reviewer: Good to make it clear from the start that this only works for Deltares employees.

First install WSL and install docker in WSL, then create a CLI secret and log into the Deltares containers. To install docker in WSL and create a CLI secret for the following steps, follow this guide https://publicwiki.deltares.nl/display/Delft3DContainers/.

Log into Deltares containers in docker:
```bash
docker login containers.deltares.nl  # use your deltares email + token
```

You can now run the Delwaq model from this directory.
```bash
docker run --mount type=bind,source="$(pwd)/model",target=/mnt/myModel \
  --workdir /mnt/myModel containers.deltares.nl/delft3d/delft3dfm run_dimr.sh
```

> If everything worked out, there's now a new netcdf output. We'll use this to update the Ribasim model.

Reviewer: ```suggestion
If everything worked out, there's now a new netCDF output. We'll use this to update the Ribasim model.
```

> Run `parse_delwaq.py` again to update the Ribasim model with the new concentrations.

Reviewer: ```suggestion
Run `parse.py` again to update the Ribasim model with the new concentrations.
```

Why again? This looks like it is the only time.
